contracts/SimpleQuests.sol

```diff
@@ -408,29 +433,90 @@ contract SimpleQuests is Initializable, AccessControlUpgradeable {
+            questSupplies[questID] = supply;
+            questDeadlines[questID] = deadline;
+        }
+    }
+
+    // TODO: Not sure how to manage created rewards and setting the week. Maybe it should be a single function that takes in info and timestamp?
+    function addReward(ItemType rewardType, uint256 rewardRarity, uint256 rewardAmount, address rewardExternalAddress, uint256 reputationAmount) public restricted {
+        uint256 rewardID = nextRewardID++;
+        rewards[rewardID] = Reward(rewardID, rewardType, rewardRarity, rewardAmount, rewardExternalAddress, reputationAmount);
+        emit RewardAdded(rewardID);
     }
 
-    function addNewPromoQuestTemplate(Rarity tier, RequirementType requirementType, Rarity requirementRarity, uint256 requirementAmount,
-        RewardType rewardType, Rarity rewardRarity, uint256 rewardAmount, uint256 reputationAmount) public restricted {
-        uint256 questID = addNewQuest(tier, requirementType, requirementRarity, requirementAmount, rewardType, rewardRarity, rewardAmount, reputationAmount);
-        questTemplates[uint8(tier) + 10].push(questID);
+    function setWeeklyReward(uint256 id, uint256 timestamp) public restricted {
+        require(timestamp > 0, "Missing timestamp");
+        uint256 week = timestamp / 1 weeks;
+        weeklyRewards[week] = id;
+        emit WeeklyRewardSet(id, week);
     }
 
-    function addNewQuest(Rarity tier, RequirementType requirementType, Rarity requirementRarity, uint256 requirementAmount,
-        RewardType rewardType, Rarity rewardRarity, uint256 rewardAmount, uint256 reputationAmount) internal returns (uint256 questID) {
-        questID = nextQuestID++;
-        quests[questID] = Quest(questID, tier,
-            requirementType, requirementRarity, requirementAmount,
-            rewardType, rewardRarity, rewardAmount, reputationAmount);
+    // TODO: What should be the identifier here?
+    function generateRewardWeeklySeed(uint256 rewardID) assertQuestsEnabled public {
+        safeRandoms.requestSingleSeed(address(this), RandomUtil.combineSeeds(SEED_REWARD_WEEKLY, rewardID));
+    }
+
+    function claimWeeklyReward() public returns (uint256[] memory weeklyRewardIDs) {
+        uint256 currentCompletions = weeklyCompletions[msg.sender][now / 1 weeks];
+        require(weeklyRewardClaimed[msg.sender][now / 1 weeks] == false, "Reward already claimed");
+        require(currentCompletions >= vars[VAR_WEEKLY_COMPLETIONS_GOAL], "Not enough weekly completions");
+        uint256 rewardID = weeklyRewards[now / 1 weeks];
+        weeklyRewardIDs = rewardWeekly(rewardID);
+        emit WeeklyRewardClaimed(msg.sender, rewardID, weeklyRewardIDs);
+        weeklyRewardClaimed[msg.sender][now / 1 weeks] = true;
+    }
+
+    function rewardWeekly(uint256 rewardID) private returns (uint256[] memory) {
+        uint256 seed = safeRandoms.popSingleSeed(address(this), RandomUtil.combineSeeds(SEED_REWARD_WEEKLY, rewardID), true, true);
```

Comment:
Is this design intended to request seeds for the contract instead of a user?
In other instances it's characterID that's checked against having seed requested which is unique per user, but here it's rewardID which is common for all users. That would mean that one user can request seed for another one and in worst case scenario never get a reward if someone steals the seed.

Reply:
I asked @poshdan about it, because this wasn't clear to me. In this case we won't be able to use characterID, because the weekly reward is for account, not character. Dan, let me know what I should use here

Reply:
well the weekly reward is tracked per wallet, so there should be no problem passing the msg.sender as the seed owner here, since they can't just switch wallets to get another roll